php/libraries/Recording.class.inc

```diff
@@ -0,0 +1,425 @@
+ *  @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ *  @link     https://www.github.com/aces/Loris/
+ */
+namespace LORIS;
+
+/**
+ * This class defines a Recording and business logic related to Recordings.
+ *
+ * @category Recording
+ * @package  Main
+ *  @author  Cecile Madjar <cecile.madjar@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class Recording
+{
+    private $_fileid;
+
+    private $_filename;
+
+    private $_filelocation;
+
+    private $_outputtype;
+
+    private $_acquisitionmodality;
+
+    private $_filetype;
+
+    private $_centerid;
+
+    private $_entitytype;
+
+    /**
+     * Constructor
+     *
+     * @param ?int $fileid This Recording FileID
+     */
+    public function __construct(?int $fileid)
+    {
+        if ($fileid !== null) {
+            $dbrow = \NDB_Factory::singleton()
+                ->database()
+                ->pselectRow(
+                    '
+                     SELECT
+                       f.FilePath               as filelocation,
+                       pot.OutputTypeName       as outputtype,
+                       pm.PhysiologicalModality as acquisitionmodality,
+                       FileType                 as filetype,
+                       s.CenterID               as centerid,
+                       c.Entity_type            as entitytype,
+                       pc.FilePath              as channel_file_path,
+                       pe.FilePath              as electrode_file_path,
+                       pte.FilePath             as event_file_path,
+                       pa.FilePath              as archive_file_path
+                     FROM
+                       physiological_file f 
+                     LEFT JOIN session s
+                       ON (f.SessionID = s.ID)
+                     LEFT JOIN candidate c
+                       ON (s.CandID = c.CandID)
+                     LEFT JOIN  physiological_output_type pot
+                       ON (pot.PhysiologicalOutputTypeID = f.PhysiologicalOutputTypeID)
+                     LEFT JOIN physiological_modality pm
+                       ON (pm.PhysiologicalModalityID = f.PhysiologicalModalityID)
+                     LEFT JOIN (
+                       SELECT PhysiologicalFileID, FilePath
+                       FROM physiological_channel
+                       LIMIT 1
+                     ) pc
+                       ON (pc.PhysiologicalFileID = f.PhysiologicalFileID)
+                     LEFT JOIN (
+                       SELECT PhysiologicalFileID, FilePath
+                       FROM physiological_electrode
+                       LIMIT 1
+                     ) pe
+                       ON (pe.PhysiologicalFileID = f.PhysiologicalFileID)
+                     LEFT JOIN (
+                       SELECT PhysiologicalFileID, FilePath
+                       FROM physiological_task_event
+                       LIMIT 1
```

Comment:
Why do you put a limit of 1 on these sub-query? (I just want to know why)

Reply:
For one `PhysiologicalFileID` in `physiological_file`, there are many rows in `physiological_task_events` but those rows all refer to the same event filename in `physiological_task_events`. Note, the rows in `physiological_task_events` are all entries present in the filename that was read by the insert scripts.

 Since I only want to know what is the file with the events, I query only the event filename in that `physiological_task_events` table, so no need to return multiple identical rows.

Hopefully that makes sense :S

Reply:
Is `physiological_task_event`.`FilePath` the same value of `physiological_file`.`FilePath`?

Reply:
no, this is a different file that contains the events used when acquiring the recording file that is stored in physiological_file.
Same for the other tables (physiological_electrodes and physiological_channels)